app/javascript/components/modals/profile/component.jsx

```diff
@@ -18,16 +19,16 @@ class ProfileModal extends PureComponent {
   };
 
   render() {
-    const { open } = this.props;
+    const { open, filledProfile } = this.props;
 
     return (
       <Modal
-        isOpen={!!open}
+        isOpen={!!open || !filledProfile}
```

Comment:
There's one case that isn't covered and it's the one with new users creating an account:
1. Users who create a MyGFW profile (not as part of saving an area) will be taken to the MyGFW profile modal window ✅ - that's working
2. Save button will be disabled until required fields complete, but the user can exit out of form without completing required fields - ❌ - that's not working, the user cannot exit the form.

We can handle that _without_ introducing a new variable `dismissed` or anything, it's better to rely on a single source of truth. We should rely solely on `open` variable and on the component _mount_ we can set the modal open if the profile isn't filled.
So:
`<Modal isOpen={open} ... >`
and in addition add componentDidMount lifecycle method
```
componentDidMount() {
    const { filledProfile, setProfileModalOpen } = this.props;
    if (!filledProfile) {
      setProfileModalOpen(true);
    }
  }
```

Reply:
yes, we talked about this, and it's pending. I like your approach, better than using the state 🙂 thanks!